src/utils/generateDirectionalStyles.js

```diff
@@ -0,0 +1,50 @@
+import rtlCSSJS from 'rtl-css-js';
+
+function separateDirectionalStyles(originalStyles, autoRTLStyles) {
+  const sharedStyles = {};
+  const ltrStyles = { ...originalStyles };
```

Comment:
So I tried this but it seems somewhat silly because you end up needing to create it ... in pretty much every part of the loop. 
```
    .forEach(([key, value]) => {
      if (value === originalStyles[key]) {
        if (!ltrStyles) ltrStyles = { ...originalStyles };
        delete ltrStyles[key];
        sharedStyles[key] = value;
        return;
      }

      if (value && typeof value === 'object') {
        // In some cases (pseudoselectors, matchmedia queries, etc.), the style
        // value may be an object, and we need to recurse.
        const recursiveStyles = separateDirectionalStyles(originalStyles[key], value);
        if (recursiveStyles != null) {
          hasRTLStyles = true;
          const {
            sharedStyles: recursiveSharedStyles,
            ltrStyles: recursiveLtrStyles,
            rtlStyles: recursiveRtlStyles,
          } = recursiveStyles;

          if (!ltrStyles) ltrStyles = { ...originalStyles };

          sharedStyles[key] = recursiveSharedStyles;
          ltrStyles[key] = recursiveLtrStyles;
          rtlStyles[key] = recursiveRtlStyles;
        }
      } else if (value != null) {
        if (!ltrStyles) ltrStyles = { ...originalStyles };
        hasRTLStyles = true;
        rtlStyles[key] = value;
      }
    });
```
I feel like you just need it and it should be created at the top.

Reply:
You could probably wrap that logic up in a function, but yeah maybe isn't worth the squeeze. We might want to profile before making this change.